thing/index.js

```diff
@@ -514,9 +549,20 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
          // which the application will need to filter out.
          //
          if (thingShadows[thingName].persistentSubscribe === true) {
-            this._handleSubscriptions(thingName, ['update', 'get', 'delete'], ['accepted', 'rejected'],
-               'subscribe');
+            topicsObject.push({
+               operations: ['update', 'get', 'delete'],
+               statii: ['accepted', 'rejected'],
+            });
+         }
+
+         if (topicsObject.length > 0) {
+            this._handleSubscriptions(thingName, topicsObject, 'subscribe', callback);
```

Comment:
Checking granted code can be done in _handleSubscriptions() so users of it do need to check the ugly code 128 and so on. What I think it should pass to the callback is probably something like this:

callback(res, failedTopics) 

For successful subscriptions, both arguments would be null, for failed one, failed topics can be returned. For unsubscribe requests, failedTopics will always be empty.